picasso/src/main/java/com/squareup/picasso3/Request.kt

```diff
@@ -214,6 +214,13 @@ class Request internal constructor(builder: Builder) {
       val path = data.uri.toString()
       builder.ensureCapacity(path.length + KEY_PADDING)
       builder.append(path)
+
+      data.headers?.toMultimap()?.let {
+        builder
+          .append(KEY_SEPARATOR)
+          .append("headers:")
+          .append(it)
+      }
```

Comment:
i'm unsure about this.  if say an OAuth token gets refreshed for an authenticated endpoint, a subsequent image request might result in an unnecessary cache miss.